consensus/hotstuff/follower_loop.go

```diff
@@ -18,24 +19,28 @@ import (
 // Concurrency safe.
 type FollowerLoop struct {
 	*component.ComponentManager
-	log           zerolog.Logger
-	followerLogic FollowerLogic
-	proposals     chan *model.Proposal
+	log             zerolog.Logger
+	certifiedBlocks chan *model.CertifiedBlock
+	forks           Forks
 }
 
 var _ component.Component = (*FollowerLoop)(nil)
 var _ module.HotStuffFollower = (*FollowerLoop)(nil)
 
-// NewFollowerLoop creates an instance of EventLoop
-func NewFollowerLoop(log zerolog.Logger, followerLogic FollowerLogic) (*FollowerLoop, error) {
+// NewFollowerLoop creates an instance of HotStuffFollower
+func NewFollowerLoop(log zerolog.Logger, forks Forks) (*FollowerLoop, error) {
+	// We can't afford to drop messages since it undermines liveness, but we also want to avoid blocking
+	// the compliance layer. Generally, the follower loop should be able to process inbound blocks faster
+	// than they pass through the compliance layer. Nevertheless, in the worst case we will fill the
+	// channel and block the compliance layer's workers. Though, that should happen only if compliance
+	// engine receives large number of blocks in short periods of time (e.g. when catching up for).
 	// TODO(active-pacemaker) add metrics for length of inbound channels
```

Comment:
Added a medium-priority task to https://app.zenhub.com/workspaces/flow-protocol-5a2ad022cee40e25b0885f82/issues/gh/onflow/flow-go/4125 to address these metric-related TODOs.